### DISCUSSION TITLE
Clarification on project rebuild

### DISCUSSION THREAD

**Discussion by cooked** [ussserrr/stm32pio#74] [Q&A] [answered]
Hi, reading through the command help it's not clear to me how the "rebuild" works once an existing .ioc file has been modified. How do I rebuild and get the changes smoothly into the existing source code (that in the meantime has been updated, in the proper //USER sections of course)

**Comment by ussserrr** [comment, marked as answer]
`generate` command will initiate a code re-generation based on your updated .ioc file (after you've updated a hardware configuration through the CubeMX). It is important to not manually start this process from GUI but let stm32pio to do its job as the CubeMX works differently in headless and GUI modes. Pass optional `—with-build` option to also rebuild the new version of the source code (via PlatformIO).

If you've updated only your sources and not an .ioc config then just use PlatformIO directly to compile / upload (e.g. by `pio run`).
